source/MaterialXGenGlsl/Nodes/NumLightsNodeGlsl.cpp

```diff
@@ -0,0 +1,31 @@
+#include <MaterialXGenGlsl/Nodes/NumLightsNodeGlsl.h>
+
+namespace MaterialX
+{
+
+ShaderNodeImplPtr NumLightsNodeGlsl::create()
+{
+    return std::make_shared<NumLightsNodeGlsl>();
+}
+
+void NumLightsNodeGlsl::createVariables(const ShaderNode&, GenContext&, Shader& shader) const
+{
+    // Create uniform for number of active light sources
+    ShaderStage& ps = shader.getStage(HW::PIXEL_STAGE);
+    ShaderPort* numActiveLights = addStageUniform(HW::PRIVATE_UNIFORMS, Type::INTEGER, "u_numActiveLightSources", ps);
+    numActiveLights->setValue(Value::createValue<int>(0));
+}
+
+void NumLightsNodeGlsl::emitFunctionDefinition(const ShaderNode&, GenContext& context, ShaderStage& stage) const
+{
+    BEGIN_SHADER_STAGE(stage, HW::PIXEL_STAGE)
```

Comment:
It was a "clean-up" to replace static embedded shader code snippets with ShaderNodeImpl nodes, which is a direction I think we should move in.. I also added a LightSamplerNodeGlsl for the same reason, which emits the sampleLightSource function. Since these are HW specific they are not part of the general shader graph, but for longer term we might be able to move in a direction where more parts of the HW render pipe-line is described by graphs (separate from the agnostic shader graph).